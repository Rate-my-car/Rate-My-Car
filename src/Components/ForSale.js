import React, { useState, useEffect } from 'react';
import useAxios from '../hooks/useAxios'
import ReactDOM from 'react-dom'

const ForSale = (props) => {
    const [inputs, handleInputs] = useState({search: ''})
    const [carsForSale] = useAxios('/api/forsale')
    console.log(carsForSale)
    const [searchTerm, setSearchTerm] = useState('')
    const [searchResults, setSearchResults] = useState([])
    const handleChange = e => {
        setSearchTerm(e.target.value)
    }

    useEffect(() => {
        const results = carsForSale.filter(vehicle =>
            vehicle.toLowerCase().includes(searchTerm)
        )
        setSearchResults(results);
    }, [searchTerm]);

   


    return(
        <div>
            <div>
                <input 
                type = 'text'
                placeholder= 'SEARCH VEHICLES' 
                // value= {searchTerm}
                // onChange = {handleChange}
                />
                
                //  {/* <ul>
                    //  {searchResults.map(item => (
                    //      <li>{item}</li>
                    //  ))}
                //  </ul> */}
                {carsForSale.map(carsForSale => {
                    return(
                        <div>
                            <div>{carsForSale.image}</div>
                            MAKE: <div>{carsForSale.make}</div>
                            MODEL: <div>{carsForSale.model}</div>
                            YEAR: <div>{carsForSale.year}</div>
                            LOCATION:<div>{carsForSale.location}</div>
                            DESCRIPTION:<div>{carsForSale.description}</div>
                            MILAGE:<div>{carsForSale.milage}</div>
                            PRICE:<div>{carsForSale.price}</div>
                            
                            {/* OWNERINFO */}
                            <button> Details </button>
                        </div>
                    )
                })}
            </div>
            
        </div>
    )
}
export default ForSale