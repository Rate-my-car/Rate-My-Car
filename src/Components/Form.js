import React, {useState} from 'react';
import axios from 'axios'
import { withRouter } from 'react-router-dom';
import './styling/Form.scss'




const Form = (props) => {
    const [inputs, handleInputs] = useState({model: '', make: '', year: '', vin: ''})
    const postCar = () => {
        const {model, make, year, vin} = inputs
        axios.post('/api/car', {model, make, year, vin}).then((res)=> {
            props.history.push(`/vehicles/${res.data[0].car_id}`)
        })
    }
    return(
        <div className='form-container'>
            <input value={inputs.make} placeholder='Make' onChange={(e)=>handleInputs({...inputs, make: e.target.value})} />
            <input value={inputs.model} placeholder='Model' onChange={(e)=>handleInputs({...inputs, model: e.target.value})} />
            <input value={inputs.year} placeholder='Year' type='number' onChange={(e)=>handleInputs({...inputs, year: e.target.value})} />
            <input value={inputs.vin} placeholder='Vin' onChange={(e)=>handleInputs({...inputs, vin: e.target.value})} />
            <button onClick={postCar}>Add Car</button>
        </div>
    )
}
export default withRouter(Form)