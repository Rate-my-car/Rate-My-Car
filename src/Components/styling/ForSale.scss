@import "./variables";
@import "./media-queries";
@import "./typography";
@import "./buttons";
@import "./inputs";

.for-sale-container {
  display: flex;
  flex-direction: column;
  padding: 0;
  margin-bottom: 40px;
  @include small-phone {
    margin-top: 100px;
  }
  @include large-phone {
    margin-top: 100px;
  }
  @include portrait-tablet {
    margin-top: 140px;
    min-height: calc(100vh - 260px);
  }
  @include landscape-tablet {
    margin-top: 140px;
  }
  @include desktop {
    margin-top: 140px;
  }
}

.vehicles-for-sale-heading-text {
  @include small-phone {
    width: 250px;
    margin: 0 auto;
  }
  // @include large-phone{
  //     width: 540px;
  // };
  // @include portrait-tablet{
  //     width: 708px;
  // };
  // @include landscape-tablet{
  //     width: 872px;
  // };
  // @include desktop{
  //     width: 1140px;
  // };
}

.car-for-sale-container {
  margin: 25px auto;
  border: 1px solid $lighter-gray;
  text-align: left;
  position: relative;
  @include small-phone {
    width: 282px;
    padding: 17px;
  }
  @include large-phone {
    width: 507px;
    padding: 17px;
    display: flex;
  }
  @include portrait-tablet {
    width: 678px;
    padding: 17px;
  }
  @include landscape-tablet {
    width: 842px;
    padding: 17px;
  }
  @include desktop {
    width: 1110px;
    padding: 17px;
  }
}

.car-for-sale-img {
  object-fit: cover;
  @include small-phone {
    width: 282px;
    height: 132px;
    margin-bottom: 15px;
  }
  @include large-phone {
    width: 223px;
    height: 185px;
  }
  @include portrait-tablet {
    width: 223px;
    height: 185px;
  }
  @include landscape-tablet {
    width: 223px;
    height: 185px;
  }
  @include desktop {
    width: 223px;
    height: 185px;
  }
}

.car-for-sale-info-container {
  position: relative;
  @include large-phone {
    margin-left: 15px;
  }
  @include portrait-tablet {
  }
  @include landscape-tablet {
    margin-left: 20px;
  }
  @include desktop {
  }
}

.car-for-sale-title-location {
  display: flex;
  @include small-phone {
    flex-direction: column;
  }
  @include large-phone {
    flex-direction: column;
  }
  @include portrait-tablet {
    flex-direction: column;
  }
  @include landscape-tablet {
    flex-direction: row;
    justify-content: baseline;
    align-items: baseline;
    flex-wrap: wrap;
  }
  @include desktop {
  }
}

.car-for-sale-title {
  @include landscape-tablet {
    margin-right: 15px;
  }
  @include desktop {
  }
}

.car-for-sale-location {
  @include fine-print;
  @include portrait-tablet {
    flex-direction: column;
    margin: 0;
  }
  @include landscape-tablet {
  }
  @include desktop {
  }
}

.car-for-sale-description {
  @include small-phone {
    display: none;
  }
  @include large-phone {
    display: none;
  }
  @include portrait-tablet {
    display: inherit;
    margin: 0 0 10px 0;
    max-height: 90px;
    overflow: scroll;
  }
  @include landscape-tablet {
    display: inherit;
    margin: 5px 0 10px 0;
    max-height: 90px;
    overflow: scroll;
  }
  @include desktop {
  }
}

.car-for-sale-price {
  color: $red;
  @include portrait-tablet {
    position: absolute;
    bottom: 0;
  }
  @include landscape-tablet {
    position: absolute;
    bottom: 0;
  }
  @include desktop {
    position: absolute;
    bottom: 0;
  }
}

.car-for-sale-details-btn {
  position: absolute;
  right: 17px;
  bottom: 17px;
  @include red-button;
}

// @include small-phone{
//     width: 325px;
// };
// @include large-phone{
//     width: 540px;
// };
// @include portrait-tablet{
//     width: 708px;
// };
// @include landscape-tablet{
//     width: 872px;
// };
// @include desktop{
//     width: 1140px;
// };
