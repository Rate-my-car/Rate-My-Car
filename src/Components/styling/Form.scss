@import "./variables";
@import "./media-queries";
@import "./typography";
@import "./buttons";

.form-container {
  display: flex;
  flex-direction: column;
  padding: 0;
  // margin-bottom: 40px;
  @include small-phone {
    margin: 110px auto 40px auto;
    width: 300px;
  }
  @include large-phone {
    margin: 110px auto 40px auto;
    width: 300px;
  }
  @include portrait-tablet {
    margin: 140px auto 40px auto;
    min-height: calc(100vh - 260px);
    width: 300px;
  }
  @include landscape-tablet {
    margin: 140px auto 40px auto;
    width: 300px;
  }
  @include desktop {
    margin: 140px auto 40px auto;
    width: 300px;
  }
}

.form-line{
  margin: 5px auto;
}

#upload-vehicle-pic-btn{
  width: 256px;
}

.new-vehicle-pic {
  width: 190px;
  height: 175px;
  object-fit: cover;
  margin: 30px auto 10px auto;
}

.add-car-form-btns {
  display: flex;
  justify-content: space-between;
  margin-top: 20px;
}



.add-car-cancel-btn {
  @include gray-button;
}

.add-car-add-btn {
  @include red-button;
}

// @include small-phone{
//     width: 325px;
// };
// @include large-phone{
//     width: 540px;
// };
// @include portrait-tablet{
//     width: 708px;
// };
// @include landscape-tablet{
//     width: 872px;
// };
// @include desktop{
//     width: 1140px;
// };
