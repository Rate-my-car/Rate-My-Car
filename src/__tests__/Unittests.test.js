import React, { Profiler } from 'react'
// import {unmountComponentAtNode} from 'react-dom'
import renderer from 'react-test-renderer'
import {HashRouter, Link} from 'react-router-dom'
import ForSale from '../Components/ForSale'
import Form from '../Components/Form'

describe('Routes will work', ()=> {
    //todd 1
    test('will bring you to the dashboard "/"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    //todd 2
    test('will bring you to the for sale page "/ForSale"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/ForSale'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    //todd 3
    test('will bring you to the maps "/maps"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/maps'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    //todd 4
    test('will bring you to the form "/form"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/form'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    //todd 5
    test('will bring you to the profile "/profile"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/profile'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    //tia 1
    test('Will take you to the Login/Register Page "/auth"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/auth'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    //tia 2
    test('Will take you to the My Vehicles Page "/MyVehicles"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/MyVehicles'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    //tia 3
    test('Will take you to a single vehicle "/vehicles/:id"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/vehicles/:id'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    //tia 4
    test('Will take you to the Post Car for Sale Page "/postforsale"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/postforsale'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    //tia 5
    test('Will take you to the Dashboard/Landing Page "/"', ()=> {
        const component = renderer.create(
            <HashRouter>
                <Link to='/'/>
            </HashRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})