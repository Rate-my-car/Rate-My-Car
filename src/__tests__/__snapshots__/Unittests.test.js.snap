// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes will work Will take you to a single vehicle "/vehicles/:id" 1`] = `
<a
  href="#/vehicles/:id"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Dashboard Page "/auth" 1`] = `
<a
  href="#/auth"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Dashboard/Landing Page "/" 1`] = `
<a
  href="#/auth"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Dashboard/Landing Page "/auth" 1`] = `
<a
  href="#/auth"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Login/Register Page "/auth" 1`] = `
<a
  href="#/auth"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Login/Register Page "/auth" 2`] = `
<a
  href="#/auth"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Login/Register Page "/postforsale" 1`] = `
<a
  href="#/postforsale"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the My Vehicles Page "/MyVehicles" 1`] = `
<a
  href="#/MyVehicles"
  onClick={[Function]}
/>
`;

exports[`Routes will work Will take you to the Post Car for Sale Page "/postforsale" 1`] = `
<a
  href="#/postforsale"
  onClick={[Function]}
/>
`;

exports[`Routes will work will bring you to the dashboard "/" 1`] = `
<a
  href="#/"
  onClick={[Function]}
/>
`;

exports[`Routes will work will bring you to the dashboard "/" 2`] = `
<a
  href="#/"
  onClick={[Function]}
/>
`;

exports[`Routes will work will bring you to the for sale page "/ForSale" 1`] = `
<a
  href="#/"
  onClick={[Function]}
/>
`;

exports[`Routes will work will bring you to the form "/form" 1`] = `
<a
  href="#/form"
  onClick={[Function]}
/>
`;

exports[`Routes will work will bring you to the maps "/maps" 1`] = `
<a
  href="#/maps"
  onClick={[Function]}
/>
`;

exports[`Routes will work will bring you to the profile "/profile" 1`] = `
<a
  href="#/profile"
  onClick={[Function]}
/>
`;
